### ISSUE TITLE
Velero backup is getting partially failed

### ISSUE THREAD

**Issue by vishumindtree** [vmware-tanzu/velero#3012]
I am getting partially failed statuses for few backups . I have pasted below the error that we get while fetching logs of velero backups. I have used --insecure-skip-tls-verify flag as I am using customized certificates.

x509: certificate signed by unknown authority
The --insecure-skip-tls-verify flag can also be used to accept any TLS certificate for the download, but it is susceptible to man-in-the-middle attacks.

**Comment by carlisia** [comment]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)


**What did you expect to happen:**


**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Comment by carlisia** [comment]
Hey @vishumindtree, please send us information as per the report I just pasted above. Include full logs if you can, and also the full commands you used. Thanks.

**Comment by nrb** [comment]
@vishumindtree Also take a look at https://velero.io/docs/v1.5/self-signed-certificates/#trusting-a-self-signed-certificate-with-the-velero-client, as there is a way to pass custom certificates with the Velero client now.

**Comment by ashish-amarnath** [comment]
@vishumindtree Here is another reference for you to use for setting up your backup storage location with custom self-signed CA certificates. https://github.com/vmware-tanzu/velero/wiki/Contributing-FAQ#how-do-i-use-minio-with-https-using-a-custom-ca-as-an-object-store-for-velero

**Comment by ashish-amarnath** [comment]
Considering that this issue has sufficient resources to use as a reference for resolution and that we don't have actionable info to troubleshoot. I am closing this issue. Please reach out to us if you have more questions or need further assistance.

**Comment by vishumindtree** [comment]
Hi Team ,
Sorry for the delay . 

I have one query . Is it okay to have partially failed backups ? Will it create an issue while restoring this backup?

**Comment by vishumindtree** [comment]
I have been trying this on aws provider , minio storage . 
1. velero version
Client:
        Version: v1.4.2
        Git commit: 56a08a4d695d893f0863f697c2f926e27d70c0c5


2. velero backup logs -

An error occurred: Get "https://***********:9000/production/backups/sw-gpu-spark/sw-gpu-spark-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=MinIo%!B(MISSING)1oss0m%!F(MISSING)20201028%!F(MISSING)minio%!F(MISSING)s3%!F(MISSING)aws4_request&X-Amz-Date=20201028T063728Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=acea951365c4e5734034bab1f60941a8075556e4e08cfc075a478ae7463faa0c": x509: certificate signed by unknown authority

The --insecure-skip-tls-verify flag can also be used to accept any TLS certificate for the download, but it is susceptible to man-in-the-middle attacks.

**Comment by ashish-amarnath** [comment]
@vishumindtree i don't think having partially failed backups should be acceptable.

The error you are getting is because you are not using the `--cacert` flag to the `velero backup logs` command. The client needs the cacert to verify thr certificate that it is presented with at the time of the ssl handahake

**Comment by vishumindtree** [comment]
1. Yeah , if I pass cacert bundle with -- > velero backup logs clustera --cacert server.pem  , then I dont get above error , but still

velero restore fails with same error.

2. Further even etcd is not getting restored - Is etcd backup & restore of kubernetes  supported by velero ?

**Comment by vishumindtree** [comment]
devops@sc-ipp-blossom-prod-k8w-30:~$ `velero restore logs wholekubeadmcluster-20201109020734` --cacert server.pem
An error occurred: Get "https://sc-ipp-blossom-minio-01.nvidia.com:9000/local/restores/wholekubeadmcluster-20201109020734/restore-wholekubeadmcluster-20201109020734-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=MinIo%!B(MISSING)1oss0m%!F(MISSING)20201109%!F(MISSING)minio%!F(MISSING)s3%!F(MISSING)aws4_request&X-Amz-Date=20201109T100924Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=1ce7991e7fee55b6842c3d01a68330c7197b3e97e38d360456f981d7fa7de453": x509: certificate signed by unknown authority

The --insecure-skip-tls-verify flag can also be used to accept any TLS certificate for the download, but it is susceptible to man-in-the-middle attacks.
--cacert: command not found

**Comment by vishumindtree** [comment]
I am still getting partially failed status of backups --

vishakumar@VISHAKUMAR-LT1:~$ velero backup create clustec --wait
Backup request "clustec" submitted successfully.
Waiting for backup to complete. You may safely press ctrl-c to stop waiting - your backup will continue in the background.

velero backup logs clustec --cacert server.pem | grep error
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-8aa23e96-000b-496c-bf6f-7ed7e047d24b namespace= persistentVolume=pvc-8aa23e96-000b-496c-bf6f-7ed7e047d24b resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-53b06041-4140-4fb6-98c6-4b191d605723 namespace= persistentVolume=pvc-53b06041-4140-4fb6-98c6-4b191d605723 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-dbf82238-1380-41fe-a3b0-9daa897278e4 namespace= persistentVolume=pvc-dbf82238-1380-41fe-a3b0-9daa897278e4 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-c9516c40-6725-4042-b301-37707506cd98 namespace= persistentVolume=pvc-c9516c40-6725-4042-b301-37707506cd98 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-193679f3-6fee-450c-b1fe-9a02d2d6aeef namespace= persistentVolume=pvc-193679f3-6fee-450c-b1fe-9a02d2d6aeef resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-f2bea0fb-8785-40b8-982e-982b3eb89c10 namespace= persistentVolume=pvc-f2bea0fb-8785-40b8-982e-982b3eb89c10 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-4e952541-e67e-4483-820f-132ae624e2bd namespace= persistentVolume=pvc-4e952541-e67e-4483-820f-132ae624e2bd resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-dbb49f8a-7d22-456c-8dde-7982dc834a09 namespace= persistentVolume=pvc-dbb49f8a-7d22-456c-8dde-7982dc834a09 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:03Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-bf40add0-d814-4daa-94ff-d7f109dd23f7 namespace= persistentVolume=pvc-bf40add0-d814-4daa-94ff-d7f109dd23f7 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:04Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-ab4bb7b5-f392-43e5-b8f7-0c31099d10f0 namespace= persistentVolume=pvc-ab4bb7b5-f392-43e5-b8f7-0c31099d10f0 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:04Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-bf69c240-f7f5-4bc4-aa60-3ac625044282 namespace= persistentVolume=pvc-bf69c240-f7f5-4bc4-aa60-3ac625044282 resource=persistentvolumes volumeSnapshotLocation=default
time="2020-11-09T13:04:04Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:448" name=pvc-b69b6848-ee3a-4a7d-9188-c38717a56af4 namespace= persistentVolume=pvc-b69b6848-ee3a-4a7d-9188-c38717a56af4 resource=persistentvolumes volumeSnapshotLocation=default

**Comment by ashish-amarnath** [comment]
> Yeah , if I pass cacert bundle with -- > velero backup logs clustera --cacert server.pem , then I dont get above error , but still
velero restore fails with same error.

I don't see any evidence of restore operations in your logs to investigate failure to restore.


Looking at the logs that you shared above, it is `velero backup create` that is failing. The reason for the failure is in the log messages

>```"Error getting volume snapshotter for volume snapshot location" backup=velero/clustec error="rpc error: code = Unknown desc = missing region in aws configuration"```

**Comment by djfoley01** [comment]
was there ever a solution to this? I'm experiencing the exact same issue. Installed with this:

velero install \
   --provider aws \
   --use-restic \
   --plugins velero/velero-plugin-for-aws:v1.1.0 \
   --bucket velero \
   --cacert ./dev-signer.crt \
   --secret-file ./minio-creds \
   --backup-location-config region=minio,s3ForcePathStyle=true,s3Url=https://s3.apps.openshiftcluster.example.domain
   
 I too am still experiencing:
 $ velero backup logs nginx-example-backup2
An error occurred: Get "https://s3.apps.openshiftcluster.example.domain/velero/backups/nginx-example-backup2/nginx-example-backup2-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=8bfe86cf-1d30-491a-b0fb-c6cac8454396%!F(MISSING)20201228%!F(MISSING)minio%!F(MISSING)s3%!F(MISSING)aws4_request&X-Amz-Date=20201228T172012Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=077abb6981f2995f83304bd065bc5429c2e14a649e556ad5a61f3dc41f7d0840": x509: certificate signed by unknown authority

The --insecure-skip-tls-verify flag can also be used to accept any TLS certificate for the download, but it is susceptible to man-in-the-middle attacks.



I know for a fact the cert is good, because I have used it elsewhere. Also after I added it, the restic pods finally stopped crashing with a tls error. This only occurs when it tries to backup the persistent volume. 
Minio Data:
$ mc ls dev/velero/backups/nginx-example-backup2 --insecure
[2020-12-28 11:19:25 CST]    29B nginx-example-backup2-csi-volumesnapshotcontents.json.gz
[2020-12-28 11:19:25 CST]    29B nginx-example-backup2-csi-volumesnapshots.json.gz
[2020-12-28 11:19:25 CST] 6.7KiB nginx-example-backup2-logs.gz
[2020-12-28 11:19:25 CST]    29B nginx-example-backup2-podvolumebackups.json.gz
[2020-12-28 11:19:25 CST]   242B nginx-example-backup2-resource-list.json.gz
[2020-12-28 11:19:25 CST]    29B nginx-example-backup2-volumesnapshots.json.gz
[2020-12-28 11:19:25 CST] 4.7KiB nginx-example-backup2.tar.gz
[2020-12-28 11:19:25 CST] 2.3KiB velero-backup.json

Persistent storage is actually missing.
